python/ray/tests/test_environ.py

```diff
@@ -0,0 +1,41 @@
+import os
+import pytest
+import ray
+
+
+@pytest.mark.skipif("sys.platform != 'linux'")
+def test_environ_file_on_linux(ray_start_10_cpus):
+    @ray.remote
+    class Actor1:
+        def __init__(self):
+            pass
+
+        def get_env_from_proc(self):
+            pid = os.getpid()
+            env = {}
+            with open("/proc/%s/environ" % pid) as fd:
+                for envspec in fd.read().split("\000"):
+                    if not envspec:
+                        continue
+                    varname, varval = envspec.split("=", 1)
+                    env[varname] = varval
+            return env
+
+        def get_os_environ(self):
+            return os.environ
+
+    a = Actor1.remote()
+    actor_proc_environ = ray.get(a.get_env_from_proc.remote())
+    print(len(actor_proc_environ))
+    actor_os_environ = ray.get(a.get_os_environ.remote())
+    assert len(actor_proc_environ) > 0
+    assert len(actor_os_environ) > 0
```

Comment:
Can we gurantee all of environ variables in os.environ equal to results from /proc/$PID/environ?

Reply:
There is no gurantee for that. /proc/$PID/environ will not reflect os.environ changes